kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/RestKieServerControllerImpl.java

```diff
@@ -49,7 +49,24 @@
 
     private static final Logger logger = LoggerFactory.getLogger(RestKieServerControllerImpl.class);
 
-    private KieServerControllerStorage storage = InMemoryKieServerControllerStorage.getInstance();
+    @POST
+    @Path("server/{serverInstanceId}")
+    @Consumes({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+    public Response updatetKieServer(@Context HttpHeaders headers,
+                                     @PathParam("serverInstanceId") String id,
+                                     String serverInfoPayload) {
+        String contentType = getContentType(headers);
+        logger.debug("Received connect request from server with id {}", id);
+        KieServerStateInfo kieServerStateInfo = unmarshal(serverInfoPayload, contentType, KieServerStateInfo.class);
+
+        logger.debug("Server info update {}", kieServerStateInfo);
+        KieServerSetup serverSetup = update(kieServerStateInfo);
+
+        String response = marshal(contentType, serverSetup);
+        logger.debug("Returning response for connect of server '{}': {}", id, response);
+        return createCorrectVariant(response, headers, serverSetup.hasNoErrors() ? Response.Status.ACCEPTED : Response.Status.BAD_REQUEST);
```

Comment:
There's a problem when returning ACCEPTED, because in the client side, when that response is processed at _makeHttpPostRequestAndCreateCustomResponse_ method, is considered as an exception:

_Exception encountered while syncing with controller at http://localhost:8080/business-central/rest/controller/server/test9-rest-2965 error Error while sending POST request to http://localhost:8080/business-central/rest/controller/server/test9-rest-2965 response code 202_

So, add ACCEPTED to the expected results in the _makeHttpPostRequestAndCreateCustomResponse_ (not sure if will affect other tests) or return OK (200) in this scenario.

Reply:
done